tests/test_load.py

```diff
@@ -728,6 +732,74 @@ def test_load_table(version, table):
     assert files == expected_files
 
 
+@pytest.mark.parametrize(
+    "version, table, map, expected",
+    [
+        (
+            "1.0.0",
+            "files",
+            None,
+            pd.concat(
+                [
+                    pd.Series(
+                        index=audformat.filewise_index(
+                            [f"audio/00{n + 1}.wav" for n in range(5)]
+                        ),
+                        name="speaker",
+                        data=["adam", "adam", "eve", "eve", None],
```

Comment:
So fully crossing the the type definitions is not necessary I think.

I mean somethig like a value that does not occur:

```
data=["john", "adam", "eve", "eve", None],
dtype=pd.CategoricalDtype(["adam", "eve"]),
```

or a categorical type having a category that is not in the data:

```
data=["adam", "adam", "eve", "eve", None],
dtype=pd.CategoricalDtype(["john", "adam", "eve"]),
```

Reply:
The former will not work as you are not allowed to set `"john"` if it is not specified in the labels.

The later works, and occurs in line 784 (of the same test).

Reply:
Yes the fomer fails but I would have expected a different reason - as you say it is not specified in the labels. 

```
                        data=["john", "adam", "eve", "eve", None],
                        dtype=pd.CategoricalDtype(["adam", "eve"]),
```

I get an AssertionError `DataFrame.iloc[:, 0] (column name="speaker") are different`. Should it not be a  pandas error related to `pd.CategoricalDtype`?



```
tests/test_load.py:801: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
testing.pyx:55: in pandas._libs.testing.assert_almost_equal
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   AssertionError: DataFrame.iloc[:, 0] (column name="speaker") are different

```

Reply:
You are specifying the expected data with
```
                        data=["john", "adam", "eve", "eve", None],
                        dtype=pd.CategoricalDtype(["adam", "eve"]),
```
not what you are requesting. So it fails as `"john"` is not in the returned data.
I was referring that it will fail if you try to create a column with such a scheme and then try to assign the labels (but this is not really related to the test we are doing here):

```python
import audformat
db = audformat.Database("db")
db.schemes["speaker"] = audformat.Scheme("str", labels=["adam", "eve"])
db["files"] = audformat.Table(audformat.filewise_index(["f1", "f2"]))
db["files"]["speaker"] = audformat.Column(scheme_id="speaker")
```
and the:
```python
>>> db["files"]["speaker"].set(["adam", "john"])
...
ValueError: Some value(s) do not match scheme
dtype: str
labels: [adam, eve]
with scheme ID 'speaker':
'john'
```
and it will also fail if you try to assign to the requested dataframe:
```python
>>> db["files"]["speaker"].set(["adam", None])
>>> df = db["files"].get()
>>> df["speaker"]
file
f1    adam
f2     NaN
Name: speaker, dtype: category
Categories (2, object): ['adam', 'eve']
>>> df["speaker"].iloc[0, 0] = "john"
...
TypeError: Cannot setitem on a Categorical with a new category (john), set the categories first
```

Reply:
Ah my bad. I was manipulating the expected data. I misread the test paramtrization. Sorry for that. 
So this conversation can be resolved. I think there is nothing more then and I should be able to proceed with approval.